internal/libyaml/reader.go

```diff
@@ -364,16 +364,15 @@ func (parser *Parser) updateBuffer(length int) bool {
 			}
 
 			// Check if the character is in the allowed range:
-			//      #x9 | #xA | #xD | [#x20-#x7E]               (8 bit)
-			//      | #x85 | [#xA0-#xD7FF] | [#xE000-#xFFFD]    (16 bit)
-			//      | [#x10000-#x10FFFF]                        (32 bit)
+			// For JSON compatibility in quoted scalars, we must allow all
+			// non-C0 characters. This includes ASCII DEL (0x7F) and the
+			// C1 control block [#x80-#x9F].
+			// ref: https://yaml.org/spec/1.2.2/#51-character-set
 			switch {
 			case value == 0x09:
 			case value == 0x0A:
 			case value == 0x0D:
-			case value >= 0x20 && value <= 0x7E:
-			case value == 0x85:
-			case value >= 0xA0 && value <= 0xD7FF:
+			case value >= 0x20 && value <= 0xD7FF:
 			case value >= 0xE000 && value <= 0xFFFD:
 			case value >= 0x10000 && value <= 0x10FFFF:
```

Comment:
Also, I feel like some of the one you add are only supported in YAML 1.2 spec, and not by previous specs. So maybe you could add an information about it in comment for each line you add. It would be easier later to enable/disable them via an option such as allowing YAML 1.1 strict only in the parsing.